Classes/Driver/Version6/IndexerDriver.php

```diff
@@ -126,7 +126,7 @@ public function fulltext(NodeInterface $node, array $fulltextIndexOfNode, string
                             ctx._source.neos_fulltext_parts = new HashMap();
                         }
 
-                        if (params.nodeIsRemoved || params.nodeIsHidden || params.fulltext.size() == 0) {
```

Comment:
If content from hidden nodes are added to the fulltext_parts, I assume they are automatically merged to the parent document where it is impossible to distinguish between content from hidden and not hidden node and the document would also be found by the originally hidden content.